### ISSUE TITLE
Build swagger skeleton for API of Group 1

### ISSUE THREAD

**Issue by felixschorer** [MusicConnectionMachine/api#9]
Build a swagger skeleton similar to #7

**Comment by felixschorer** [comment]
@MusicConnectionMachine/group-1 I am more than happy to start implement that right now. #7 needs to be merged first though to avoid conflicts.

**Comment by felixschorer** [comment]
There's not much else I could do right now as I don't have an overview over the database structure.

**Comment by felixschorer** [comment]
I'll start working on it then. Thanks for merging the pull request, @kordianbruck.

**Comment by kordianbruck** [comment]
@lgfrbcsgo awesomooooo - keep it going :dancer:

**Comment by felixschorer** [comment]
It's surprisingly fun to work with swagger once you get the hang of it 😄

**Comment by kordianbruck** [comment]
@lgfrbcsgo cause it has swag :drum: ba-dum-tsssss

**Comment by felixschorer** [comment]
Wow, `swagger.yaml` is growing quickly in size. I haven't found a way to split it into multiple files and apperently it's currently not possible with swagger.

@MusicConnectionMachine/group-1 are you sure that point `1.2` in the Google Doc is really necessary? Would something like `/entities?type=artist` work?

**EDIT:** I'll try some more things. Just wanted to warn you in case `swagger.yaml` blows up

**Comment by kordianbruck** [comment]
@lgfrbcsgo will this not work? http://azimi.me/2015/07/16/split-swagger-into-smaller-files.html

**Comment by felixschorer** [comment]
I'll give it a try. Was searching for first party solutions, @kordianbruck.

**Comment by kordianbruck** [comment]
@lgfrbcsgo finish the endpoints for group 1 first please, so shilpa can start porting the code. Thanks!

**Comment by felixschorer** [comment]
The skeleton for the `/entities` path is finished. Shall I create a pull request and add the other paths later? Got the compiling to `.json` working. @kordianbruck

**Comment by felixschorer** [comment]
Pull request is up @kordianbruck @MusicConnectionMachine/group-1

**Comment by kordianbruck** [comment]
@lgfrbcsgo thanks a lot! I hope group one appreciates this as well! @TimHenkelmann @LukasNavickas @angelinrashmi2000 @ShilpaGhanashyamGore

**Comment by TimHenkelmann** [comment]
First of all, thanks a lot for your swagger implementation of the /entities path. But after todays discusssion, it seems like we do not need this entities table any longer. So the API's agreed for /entities is no longer valid. Instead, we have come up with a new set of API's listed below. Therefore, we will need to create a new swagger specifiation file for these. 
@MusicConnectionMachine/group-2 @MusicConnectionMachine/group-3  @MusicConnectionMachine/group-4  @MusicConnectionMachine/group-5  @MusicConnectionMachine/group-6 Please have a look at it and let us know if you have any concerns, otherwise we will implement it like that.

New API (this is also added to the googledrive):

GET/artists/
GET/artists/artist_id
POST/artists
PUT/artists/artist_id
DELETE/artists/artist_id

GET/works/
GET/works/work_id
POST/works
PUT/works/work_id
DELETE/works/work_id

GET/releases/
GET/releases/release_id
POST/releases
PUT/releases/release_id
DELETE/releases/release_id

GET/instruments/
GET/instruments/instrument_id
POST/instruments
PUT/instruments/instrument_id
DELETE/instruments/instrument_id

GET/works?artist_id=id
GET/releases?artist_id=id
GET/artists?work_id=id
GET/artists?release_id=id
GET/artists?instrument_id=id
GET/instruments?artist_id=id

**Comment by metavaults** [comment]
I have updated the API doc

**Comment by felixschorer** [comment]
`/artists`, `/works`, `/releases` and `/instruments` should be pretty straight forward. Basically just copying `/entities` and changing descriptions and handler functions. 
Just tag me if you need help with anything. @MusicConnectionMachine/group-1 
The optional parameters work just like the `apiKey` parameters.

**Comment by felixschorer** [comment]
I can also join a Hangouts session if needed.

**Comment by kordianbruck** [comment]
So this is finished? @TimHenkelmann

**Comment by ShilpaGhanashyamGore** [comment]
Yes we have completed this task
